pkg/cloud-controller-manager/loadBalancer.go

```diff
@@ -70,18 +67,8 @@ func (l *LoadBalancerManager) GetLoadBalancer(ctx context.Context, clusterName s
 }
 
 func (l *LoadBalancerManager) GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string {
-	// If users don't specify cluster name by the command flag, we adopt the cluster name from the cloud config file
-	if clusterName == "" || clusterName == "kubernetes" {
-		clusterName = l.clusterName
-	}
-	name := clusterName + "-" + service.Namespace + "-" + service.Name
-	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names
-	// The name contains no more than 253 characters
-	if len(name) > 253 {
-		name = name[:253]
-	}
-
-	return name
+	// refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/names
+	return "a" + string(service.UID) + "a"
```

Comment:
use generate name to create the unique LB name and add service UID to the annotation

Reply:
You have the same view as the Kubernetes code developer. :thumbsup:
```
// (DEPRECATED) DefaultLoadBalancerName is the default load balancer name that is called from
// LoadBalancer.GetLoadBalancerName. Use this method to maintain backward compatible names for
// LoadBalancers that were created prior to Kubernetes v1.12. In the future, each provider should
// replace this method call in GetLoadBalancerName with a provider-specific implementation that
// is less cryptic than the Service's UUID.
func DefaultLoadBalancerName(service *v1.Service) string {
	//GCE requires that the name of a load balancer starts with a lower case letter.
	ret := "a" + string(service.UID)
	ret = strings.Replace(ret, "-", "", -1)
	//AWS requires that the name of a load balancer is shorter than 32 bytes.
	if len(ret) > 32 {
		ret = ret[:32]
	}
	return ret
}
```